pkg/cli/image/strategy/explicit_test.go

```diff
@@ -0,0 +1,221 @@
+				{
+					Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{
+							{
+								Source: "docker.io/ocp-test/does-not-exist",
+								Mirrors: []string{
+									"does.not.exist/match/image",
+								},
+							},
+							{
+								Source: "quay.io/ocp-test/does-not-exist",
+								Mirrors: []string{
+									"exists/match/image",
+								},
+							},
+						},
+					},
+				},
+			},
+			image:                "quay.io/ocp-test/does-not-exist:4.7",
+			imageSourcesExpected: []string{"exists/match/image", "quay.io/ocp-test/does-not-exist"},
+		},
+		{
+			name: "single mirror and match",
+			icspList: []operatorv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{
+							{
+								Source: "quay.io/ocp-test/release",
+								Mirrors: []string{
+									"someregistry/mirrors/match",
+								},
+							},
+						},
+					},
+				},
+			},
+			image:                "quay.io/ocp-test/release:4.5",
+			imageSourcesExpected: []string{"someregistry/mirrors/match", "quay.io/ocp-test/release"},
+		},
+		{
+			name: "no source match",
+			icspList: []operatorv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{
+							{
+								Source: "docker.io/ocp-test/does-not-exist",
+								Mirrors: []string{
+									"does.not.exist/match/image",
+								},
+							},
+							{
+								Source: "quay.io/ocp-test/does-not-exist",
+								Mirrors: []string{
+									"exists/match/image",
+								},
+							},
+						},
+					},
+				},
+			},
+			image:                "quay.io/passed/image:4.5",
+			imageSourcesExpected: []string{"quay.io/passed/image"},
+		},
+		{
+			name: "multiple mirrors for single source match",
+			icspList: []operatorv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{
+							{
+								Source: "quay.io/ocp-test/release",
+								Mirrors: []string{
+									"someregistry/mirrors/match",
+									"quay.io/another/release",
+									"quay.io/andanother/release",
+								},
+							},
+						},
+					},
+				},
+			},
+			image:                "quay.io/ocp-test/release:4.5",
+			imageSourcesExpected: []string{"someregistry/mirrors/match", "quay.io/another/release", "quay.io/andanother/release", "quay.io/ocp-test/release"},
+		},
+		{
+			name: "docker.io vs registry-1.docker.io",
+			icspList: []operatorv1alpha1.ImageContentSourcePolicy{
+				{
+					Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
+						RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{
+							{
+								Source: "docker.io/ocp-test/release",
```

Comment:
Add a test case where source mirrors include both `index.docker.io` and `registry-1.docker.io`.